kernel/src/syscall/preadv.rs

```diff
@@ -21,9 +21,14 @@ pub fn sys_preadv(
     fd: FileDesc,
     io_vec_ptr: Vaddr,
     io_vec_count: usize,
-    offset: i64,
+    offset_low: u64,
+    _offset_high: u64,
     ctx: &Context,
 ) -> Result<SyscallReturn> {
+    // On 64-bit platforms, Linux assumes the `offset_low` contains the full
+    // 64-bit offset.
+    // Reference: <https://elixir.bootlin.com/linux/v6.16.9/source/fs/read_write.c#L1114-L1118>.
+    let offset = offset_low as i64;
```

Comment:
Oh, sorry, I should have mentioned this earlier and it should be fixed together with the reference format. It should be:
```rust
    let offset = offset_low.cast_signed();
```

In general, `as` is too powerful and may truncate or extend bits incorrectly. Therefore, `cast_signed` is preferred, as it guarantees the conversion of an unsigned to a signed value.